### ISSUE TITLE
Correlation coefficient as an output from autoRIFT + destriping

### ISSUE THREAD

**Issue by arimue** [nasa-jpl/autoRIFT#43]
Hi all, 

I am attempting to use autoRIFT for offset tracking across a slow-moving landslide using optical satellite imagery. autoRIFT is running fine, however, I get a lot of unnaturally high offsets in high-relief areas which I assume are caused by shadow effects and are only partially removed by the NDC filter. Here is an example from correlating two SPOT-5 scenes:
![Offset_example](https://user-images.githubusercontent.com/33640758/139720029-c3e25eed-8d68-47e4-98e1-c467452e2dfb.png)

I have thus been thinking about alternative ways to filter the estimated pixel displacement and was wondering if it would be possible to retrieve the correlation coefficient for every matched template as an output from autoRIFT to assess how reliable each match is?

Another problem that occurs when I use Landsat 5 imagery as an input is that I get a regular stripe pattern in the offset imagery, especially for the horizontal pixel displacement.
![Offset_example2](https://user-images.githubusercontent.com/33640758/139720407-f4e5342d-7479-4acd-bc36-3bbd6de45192.png)

Have you had similar experiences or perhaps any suggestions on how to deal with this in autoRIFT? I have attempted to destripe the output by subtracting the mean in the line direction of the image, similar to what was described by Scherler et al. (2008) (doi:10.1016/j.rse.2008.05.018), but I wonder if there is a different way to approach this.

**Comment by alex-s-gardner** [comment]
Hi  @arimue arimue 

Glad you've got autoRIFT up and running and are finding it useful and thanks a ton for posting the figures... that helps a ton with diagnosing issues.So to your questions:

1. It looks to me like the two images that you are correlating have some orthorectification issues... you can clearly see residual parallax in the offset map... if you plotted the offset as a function of slope and aspect (https://tc.copernicus.org/articles/5/271/2011/) you'll probably see a strong relation. Improved filtering will not help. Also the patterns of offset are very coherent in space which tells me that the offsets are real, not spurious.

2. The striping is due to banding in the input imagery... for glaciers is handled this issue using and fft decomposition and removing power in the along-track direction then reconstructing the image but we have not yet implimented this in autoRIFT (on the to-do list). A simple fix might be simply to test alternate image bands. 

Hope that helps.

Alex

**Comment by arimue** [comment]
Thank you so much for all the suggestions, @alex-s-gardner. I will take another look at orthorectification. However, would it generally be possible to get the correlation coefficient as an output from autoRIFT, or do you just not think that it would be a good filter criterion?

**Comment by alex-s-gardner** [comment]
@arimue using correlation coefficient is a totally logical thing to suggest. We played around with using the correlation coefficient as a filter quite a bit in the early days but found that the NDC filter was far superior and including information on the correlation coefficient had little added value. For this reason we decided not to export the correlation coefficient. The CC is buried deep in the code (CPP implementation of the NCC kernel) so it would take considerable work to pass this back to the user.
